PropertyView bugfixes

mpsonntag opened the pull request:
This PR fixes a bunch of bugs in the PropertyView:
- Fixes that when adding a Terminology Property via the context menu in the PropertyView, the added Property was not initialized with pseudo_values, leaving these Properties broken with respect to adding actual values to them. Closes #161.
- Fixes that when using the popup menu to reset a Terminology Property to its merged default, the cloned Property replacing it was not properly initialized with pseudo_values. Closes #162.
- Fixes that when a Terminology Property provides pre-defined Values via the popup menu item 'Add Value' and 'Set Value', only empty values were able to be set. Actual values lead to a silent background error. Closes #163.
- Fixes that the gtk.view became out of sync with the underlying data model, virtually breaking the displayed treemodel rendering it unusable for the selected Property until another section was selected. Closes #164.
- The main `editor` now catches exceptions on `undo` to ensure, that the view is properly reset even if an error occurs.
- Does some code cleanup on the side and updates docstrings.

mpsonntag committed:
[propertyView] General code cleanup

- remove unused variables
- no redefining of variable names
- stick to snake case variable names

mpsonntag committed:
Refactor view 'on_get_tooltip' method

The 'on_get_tooltip' method on property-
and section view is now static and has
one unused method parameter removed.

mpsonntag committed:
[propertyView] Refactor dtype_renderer method

mpsonntag committed:
[propertyView] Substitute lambda

mpsonntag committed:
[propertyView] Replace outdated lambda

The 'value_filter' lambda in method 'get_popup_menu_items'
was still referring to an old version of Values.
It was replaced with the updated '_value_filter'
static method.

mpsonntag committed:
[propertyView] Properly initialize new term prop

Closes #161

A terminology property added via the popup menu
was not properly initialized with pseudo_values.

mpsonntag committed:
[propertyView] Remove unused global variables

mpsonntag committed:
[propertyView] Initialize default reset prop

mpsonntag committed:
[propertyView] Fix set value item popup lambda

Closes #162

The lambda providing default values for a
terminology Property with defined Values
was still using the previous Values model.
The changes fix the ensuing exception.

mpsonntag committed:
[propertyView] Fix 'add_value' from terminology

Fixes a bug that silently fails when trying to
add a non empty value.

mpsonntag committed:
[valueModel] Add index property

mpsonntag committed:
[propertyView] Fix 'set_value'

Closes #163

Values can now be replaced by empty
values or Values from a terminology.

mpsonntag committed:
[propertyView] Properly check for obj instance

mpsonntag committed:
[editor] Catch exceptions on undo

mpsonntag committed:
[propertyView] Fix view reset bug

Closes #164

Ensures, that the model is always created
from the parent section and that only
existing properties are selected in
the tree_view.

mpsonntag committed:
[propertyView] Remove unused import

mpsonntag committed:
[propertyView] Simplify 'on_object_edit'

mpsonntag committed:
[propertyView] Update docstrings

achilleas-k commented on odmlui/editor.py:
```diff
@@ -1091,7 +1091,11 @@ def enable_redo(self, enable=True):
+            self.current_tab.command_manager.undo()
+        except Exception as exc:
+            self._info_bar.show_info("Unable to undo last action")
+            print("Encountered and exception during undo: %s:%s" % (type(exc), exc))
```
```suggestion
            print("Encountered an exception during undo: %s:%s" % (type(exc), exc))
```

achilleas-k commented on odmlui/property_view.py:
```diff
@@ -328,28 +401,43 @@ def add_property(self, widget, obj_prop_pair):
         """
         obj = self.get_selected_object()
-        if obj is None or not isinstance(obj, TreeModelProperty):
+        if obj is None or (not hasattr(obj, "parent") and obj.parent):
```
Something's not right here.

mpsonntag committed:
[propertyView] Fix 'reset_value_view' conditional

mpsonntag committed:
[Travis] Allow failure on conda builds

Travis constantly fails with http get
errors during the conda installation.
Setting these builds to allow failure
until the issue can be resolved.

achilleas-k committed:
Update odmlui/editor.py

Co-Authored-By: mpsonntag <michael.p.sonntag@gmail.com>
